### PR TITLE
Fixed bug where section headers wouldn't move when cells did

### PR DESCRIPTION
**Pull request by maxgoedjen**
TUITableView's _layoutSectionHeaders: now checks __isDraggingCell before calling setNeedsLayout.

### CODE REVIEW

File: lib/UIKit/TUITableView.m [hunk 1]
```diff
@@ -765,8 +765,10 @@ - (void)_layoutSectionHeaders:(BOOL)visibleHeadersNeedRelayout
 					}
 				}
 				
-				section.headerView.frame = headerFrame;
-				[section.headerView setNeedsLayout];
+                section.headerView.frame = headerFrame;
```
Comment (jwilling) [inline lib/UIKit/TUITableView.m]: Can you double-check your indentation? Looks like you've got double tabs going on here.

Comment (maxgoedjen) [inline lib/UIKit/TUITableView.m]: Fixed.

Comment (jwilling) [inline lib/UIKit/TUITableView.m]: I'm not entirely sure what exactly the problem is. Definitely I see an issue with moving cells between different sections in the current demo, but I also do not see much of a difference with these changes in place. What exactly is the issue here?

In the demo, there are always 25 cells in each section, but in a real application the number will naturally change when the cells are dragged between different sections. Does your patch resolve the problem in this case?

**Comment by jwilling** [comment]
Yes, that's better. There still is a remaining problem, though. Try dragging a cell to the last row right before a section header. It is not made clear whether it's droppable there, because no gap is made between the last cell and the next section.

File: lib/UIKit/TUITableView.m (hunk 1 above)
Comment (maxgoedjen) [inline lib/UIKit/TUITableView.m]: You are correct, that was supposed to check before frame assignment too. Must've undone when I was playing with it. It works correctly in scenarios where there are dynamic numbers of cells; I'm testing it with a different app.

**Comment by maxgoedjen** [comment]
Just noticed that myself. I think it's an issue in _updateDraggingCell:offset:location:. Looking into it now.

**Comment by avaidyam** [comment]
Any updates on this? :)

**Comment by maxgoedjen** [comment]
Going to close this out and make this into a nicer branch without unneeded commits.

**Comment by avaidyam** [comment]
You should instead squash the changes.... saves time.

**Pull request closed without merging**